src/ecs/storage/Storage.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+** EPITECH PROJECT, 2022
+** Bomberman: Global Offensive
+** File description:
+** ECS - The Component Storage Trait
+*/
+
+#ifndef ECS_STORAGE_STORAGE_HPP_
+#define ECS_STORAGE_STORAGE_HPP_
+
+#include "ecs/Entity.hpp"
+#include "ecs/join/Joinable.hpp"
+
+#include <concepts>
+#include <iterator>
+
+// forward declaration of MapStorage for use as default storage type.
+namespace ecs
+{
+    template <typename C> class MapStorage;
+}
+
+/// @note Use ecs::get_storage_type instead.
+template <typename C> struct GetComponentStorageType {
```

Comment:
Is there a reason why the struct isn't in the namespace ecs

Reply:
yes, there is